WordPress/src/main/java/org/wordpress/android/ui/posts/editor/EditorActionsProvider.kt

```diff
@@ -0,0 +1,105 @@
+
+@Reusable
+class EditorActionsProvider @Inject constructor(private val remoteLoggingUtils: CrashLoggingUtilsWrapper) {
+    fun getPrimaryAction(postStatus: PostStatus, userCanPublish: Boolean): PrimaryEditorAction {
+        return if (userCanPublish) {
+            when (postStatus) {
+                PostStatus.SCHEDULED -> PrimaryEditorAction.SCHEDULE
+                PostStatus.DRAFT -> PrimaryEditorAction.PUBLISH_NOW
+                PostStatus.PENDING, PostStatus.TRASHED -> PrimaryEditorAction.SAVE
+                PostStatus.PRIVATE, PostStatus.PUBLISHED, PostStatus.UNKNOWN -> PrimaryEditorAction.UPDATE
+            }
+        } else {
+            // User doesn't have publishing permissions
+            when (postStatus) {
+                PostStatus.DRAFT,
+                PostStatus.PENDING,
+                PostStatus.UNKNOWN -> PrimaryEditorAction.SUBMIT_FOR_REVIEW
+                PostStatus.TRASHED -> {
+                    // TODO if this log won't appear in Sentry, we should start throwing IllegalStateException
+                    //  instead of returning a valid action.
+                    remoteLoggingUtils.log(
+                            "User shouldn't be able to open a trashed post in an editor " +
+                                    "without publishing rights."
+                    )
+                    PrimaryEditorAction.SAVE
+                }
+                PostStatus.PUBLISHED,
+                PostStatus.SCHEDULED,
+                PostStatus.PRIVATE -> {
+                    // TODO if this log won't appear in Sentry, we should start throwing IllegalStateException
+                    //  instead of returning a valid action.
+                    remoteLoggingUtils.log(
+                            "User shouldn't be able to open a public ($postStatus) post in an editor " +
+                                    "without publishing rights."
+                    )
+                    PrimaryEditorAction.SUBMIT_FOR_REVIEW
+                }
+            }
+        }
+    }
+
+    fun getSecondaryAction(
+        postStatus: PostStatus,
+        userCanPublish: Boolean
+    ): SecondaryEditorAction {
+        return if (userCanPublish) {
+            when (postStatus) {
+                PostStatus.DRAFT -> SecondaryEditorAction.SAVE
+                PostStatus.PENDING, PostStatus.SCHEDULED -> SecondaryEditorAction.PUBLISH_NOW
+                PostStatus.PRIVATE, PostStatus.PUBLISHED -> SecondaryEditorAction.NONE
+                PostStatus.TRASHED, PostStatus.UNKNOWN -> SecondaryEditorAction.SAVE_AS_DRAFT
+            }
+        } else {
+            // User doesn't have publishing permissions
+            when (postStatus) {
+                PostStatus.DRAFT,
+                PostStatus.PENDING,
+                PostStatus.UNKNOWN -> SecondaryEditorAction.NONE
+                PostStatus.TRASHED -> {
+                    // TODO if this log won't appear in Sentry, we should start throwing IllegalStateException
```

Comment:
Another English nitpick 😉 Instead of saying "if this log won't appear" it's better to say "if this log doesn't appear" 
won't = is final
doesn't = it might or might not happen.

Reply:
Fixed in 2bd574b13e755a27c0ad280ed27305f757841657